### PR TITLE
feat: Customers module

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

image

Screenshot 2024-01-09 at 11 38 03 AM

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/Customers/Customers.res [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      let err = Js.Exn.message(e)->Belt.Option.getWithDefault("Failed to Fetch!")
+      setScreenState(_ => PageLoaderWrapper.Error(err))
+    }
+  }
+  React.useEffect0(() => {
+    getCustomersList()->ignore
+    None
+  })
+
+  <ErrorBoundary>
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/Customers/Customers.res:29]: why <ErrorBoundary> is added over here

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Customers/Customers.res:29]: It's just a standard and we should add in every individual components that if some error happened in Customers Module the whole app won't get crashed.

Comment (JeevaRamu0104) [reply src/screens/HyperSwitch/Customers/Customers.res:29]: Expect api error we don't expect any error in the individual components.And also  <ErrorBoundary> is added at the App level right?

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Customers/Customers.res:29]: Yes it is added in the main entry component. So then we can remove from everywhere. Like Table tabs analytics and every component

Comment (JeevaRamu0104) [reply src/screens/HyperSwitch/Customers/Customers.res:29]: If its added in the main entry component then we can remove from here right

File: src/screens/HyperSwitch/Customers/Customers.res [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  open CustomersEntity
+  let fetchDetails = useGetMethod()
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
+  let (customersData, setCustomersData) = React.useState(_ => [])
+  let (offset, setOffset) = React.useState(_ => 0)
+
+  let getCustomersList = async () => {
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let customersUrl = getURL(~entityName=CUSTOMERS, ~methodType=Get, ())
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/Customers/Customers.res:13]: please check what is limit that we need to fetch

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Customers/Customers.res:13]: Added - 10000

**Review by JeevaRamu0104 — DISMISSED** [review]

File: src/components/TableUtils.res [hunk 3]
```diff
@@ -691,7 +691,7 @@ module TableCell = {
     | Text(x) | DropDown(x) => {
         let x = x === "" ? "NA" : x
         <AddDataAttributes attributes=[("data-desc", x)]>
-          <div> {highlightedText(x, highlightText)} </div>
+          <div className="whitespace-nowrap"> {highlightedText(x, highlightText)} </div>
```
Comment (Riddhiagrawal001) [inline src/components/TableUtils.res:694]: Please check this as this was resulting in Client secret and error message overlapping in showOrderDetails

File: src/screens/HyperSwitch/Customers/CustomersEntity.res [hunk 4]
```diff
@@ -0,0 +1,96 @@
+open LogicUtils
+
+type customers = {
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/Customers/CustomersEntity.res:3]: Can we move all the types in the Entity file to a new CustomerType File ?

File: src/components/TableUtils.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/components/TableUtils.res:694]: Updated

File: src/screens/HyperSwitch/Customers/CustomersEntity.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Customers/CustomersEntity.res]: Updated

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**